src/Sulu/Component/Content/Repository/ContentRepository.php

```diff
@@ -714,9 +714,9 @@ public function resolveInternalLinkContent(
         $linkedContent = $this->find($row->getValue('internalLink'), $locale, $webspaceKey, $mapping);
         $data = $linkedContent->getData();
 
-        // properties which are in the intersection of the data and non
-        // fallback properties should be handled on the original row.
-        $properties = \array_intersect(self::$nonFallbackProperties, \array_keys($data));
+        // return value of source node instead of link destination for title and non-fallback-properties
+        $sourceNodeValueProperties = array_merge(['title'], self::$nonFallbackProperties);
```

Comment:
`array_merge` should only be used when merging array together. If we just want to add something to an array it is more efficient to use the `[]` operator:

```suggestion
        $sourceNodeValueProperties = self::$nonFallbackProperties;
        $sourceNodeValueProperties[] = 'title';
```

You can see the difference when running:

```bash
php -r '$startTime = microtime(true);

              $array = [];

              for ($i = 0; $i <= 10000; ++$i) {
                  $array = array_merge([\'title\'], $array);
              }

              $endTime = microtime(true);

              var_dump($endTime - $startTime);'

# float(0.34501719474792)
```

vs:

```bash
php -r '
              $startTime = microtime(true);

              $array = [];

              for ($i = 0; $i <= 10000; ++$i) {
                  $array[] = \'title\';
              }

              $endTime = microtime(true);

              var_dump($endTime - $startTime);'
              
# float(0.00098204612731934)
```

Using in this benchmark 100000 will make the array_merge a lot slower.

Maybe not so relevant in this case but I would always go the `[]` way when its possible and we don't need to `merge` keys together.